jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -215,139 +218,133 @@ private boolean checkImagePlatform(Image image) {
       RegistryClient registryClient, ProgressEventDispatcher progressEventDispatcher)
       throws IOException, RegistryException, LayerPropertyNotFoundException,
           LayerCountMismatchException, BadContainerConfigurationFormatException {
+    Cache cache = buildContext.getBaseImageLayersCache();
+    ImageConfiguration baseImageConfig = buildContext.getBaseImageConfiguration();
+
     ManifestAndDigest<?> manifestAndDigest =
-        registryClient.pullManifest(buildContext.getBaseImageConfiguration().getImageQualifier());
+        registryClient.pullManifest(baseImageConfig.getImageQualifier());
     ManifestTemplate manifestTemplate = manifestAndDigest.getManifest();
 
+    if (manifestTemplate instanceof V21ManifestTemplate) {
+      V21ManifestTemplate v21Manifest = (V21ManifestTemplate) manifestTemplate;
+      cache.writeMetadata(baseImageConfig.getImage(), v21Manifest);
+      return Collections.singletonList(JsonToImageTranslator.toImage(v21Manifest));
+    }
+
+    ManifestTemplate manifestList = null;
+    List<BuildableManifestTemplate> manifests = new ArrayList<>();
+    List<ContainerConfigurationTemplate> containerConfigs = new ArrayList<>();
     // If a manifest list, search for the manifests matching the given platforms.
     if (manifestTemplate instanceof V22ManifestListTemplate) {
-      ImmutableList.Builder<Image> images = ImmutableList.builder();
-
+      manifestList = manifestTemplate;
       for (Platform platform : buildContext.getContainerConfiguration().getPlatforms()) {
         manifestAndDigest =
             obtainPlatformSpecificImageManifest(
-                registryClient, (V22ManifestListTemplate) manifestTemplate, platform);
-        images.add(jsonManifestToImage(manifestAndDigest, registryClient, progressEventDispatcher));
+                registryClient, (V22ManifestListTemplate) manifestList, platform);
+
+        manifests.add((BuildableManifestTemplate) manifestAndDigest.getManifest());
+        containerConfigs.add(
+            pullContainerConfigJson(manifestAndDigest, registryClient, progressEventDispatcher));
       }
-      return images.build();
+
+    } else {
+      // V22ManifestTemplate or OciManifestTemplate
```

Comment:
What about the `OciManifestListTemplate` ?

Reply:
That's why I added the comment
```
// TODO: support OciIndexTemplate once AbstractManifestPuller starts to accept it.
```
below. (Note `OciIndexTemplate` is what you call by "OCI manifest list" here.) Supporting an OCI image index is another TODO item that you're tracking.

Reply:
Makes sense ,  I wasn't aware of the fact that `OciIndexTemplate` = `Oci manifest list`